packages/@aws-cdk/aws-ecs-patterns/lib/fargate/application-load-balanced-fargate-service.ts

```diff
@@ -60,6 +60,17 @@ export interface ApplicationLoadBalancedFargateServiceProps extends ApplicationL
    */
   readonly memoryLimitMiB?: number;
 
+  /**
+   * The amount (in GiB) of ephemeral storage to be allocated to the task. The maximum supported value is 200 GiB.
+   *
+   * NOTE: This parameter is only supported for tasks hosted on AWS Fargate using platform version 1.4.0 or later.
+   *
+   * This default is set in the underlying FargateTaskDefinition construct.
+   *
+   * @default 20
```

Comment:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-ephemeralstorage.html suggests this number should be between 21 and 200:

> The total amount, in GiB, of ephemeral storage to set for the task. The minimum supported value is 21 GiB and the maximum supported value is 200 GiB.

Would 20 not create a runtime error? And/or could there be some validation of `ephemeralStorageGiB` with an error thrown when `ephemeralStorageGiB` is less than 21. That is, produce a compile (synth) time error for earlier feedback.

Reply:
> Would 20 not create a runtime error? And/or could there be some validation of ephemeralStorageGiB with an error thrown when it is not in these bounds?

Yes, entering a value below 21 or above 200 will throw a validation exception:

https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-ecs/lib/fargate/fargate-task-definition.ts#L143-L145

The default value of 20 GiB comes into play when `ephemeralStorageGiB` is not set. By default, Fargate provides 20 GiB of ephemeral storage. That is, users only need to add `ephemeralStorageGiB` to their task definition when their task needs more than 20 GiB:

https://docs.aws.amazon.com/AmazonECS/latest/developerguide/fargate-task-storage.html

I documented the 20 GiB default value for ephemeral storage for consistency with `aws-ecs`:

https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-ecs/lib/fargate/fargate-task-definition.ts#L55-L62

You can see here that `aws-ecs` doesn't actually set the default value to 20 in the task definition, but rather sets `ephemeralStorage` to `undefined` when `ephemeralStorageGiB` is not set:

https://github.com/aws/aws-cdk/blob/fa8f2e2180d60e5621d2ae9606a3d1b2dcb681d9/packages/%40aws-cdk/aws-ecs/lib/base/task-definition.ts#L470-L472

Reply:
👍🏽 thanks for the explanation and links!

Reply:
I think that having this say `20` as the the default input and then throwing an error on that value is not the way we want to go here. I realize that the `aws-ecs` module did it that way, but I tend to think that should be updated instead of this one following it's lead. Let's add a default of none, since we're not plugging the number 20 in anywhere on our end and add to this doc that the minimum value here is 21. If you'd like to make the same update to the `aws-ecs` docs, we'd welcome that but it's certainly not mandatory.